libminifi/src/utils/Cron.cpp

```diff
@@ -0,0 +1,507 @@
+
+  if (weekday_str.size() > 2) {
+    weekday parsed_weekday{};
+    from_stream(stream, "%a", parsed_weekday);
+    if (!stream.fail() && parsed_weekday.ok() && stream.peek() == EOF)
+      return parsed_weekday;
+  } else {
+    unsigned weekday_num;
+    stream >> weekday_num;
+    if (!stream.fail() && weekday_num < 7 && stream.peek() == EOF)
+      return weekday(weekday_num-1);
+  }
+  throw BadCronExpression("Invalid weekday: " + weekday_str);
+}
+
+template <>
+year parse<year>(const std::string& year_str) {
+  if (auto year_int = fromChars<uint64_t>(year_str); year_int && *year_int >= 1970 && *year_int <= 2999)
+    return year(*year_int);
+  throw BadCronExpression("Invalid year: " + year_str);
+}
+
+template <typename FieldType>
+FieldType getFieldType(local_seconds time_point);
+
+template <>
+year getFieldType(local_seconds time_point) {
+  year_month_day year_month_day(floor<days>(time_point));
+  return year_month_day.year();
+}
+
+template <>
+month getFieldType(local_seconds time_point) {
+  year_month_day year_month_day(floor<days>(time_point));
+  return year_month_day.month();
+}
+
+template <>
+day getFieldType(local_seconds time_point) {
+  year_month_day year_month_day(floor<days>(time_point));
+  return year_month_day.day();
+}
+
+template <>
+hours getFieldType(local_seconds time_point) {
+  auto dp = floor<days>(time_point);
+  auto time = make_time(time_point-dp);
+  return time.hours();
+}
+
+template <>
+minutes getFieldType(local_seconds time_point) {
+  auto dp = floor<days>(time_point);
+  auto time = make_time(time_point-dp);
+  return time.minutes();
+}
+
+template <>
+seconds getFieldType(local_seconds time_point) {
+  auto dp = floor<days>(time_point);
+  auto time = make_time(time_point-dp);
+  return time.seconds();
+}
+
+template <>
+weekday getFieldType(local_seconds time_point) {
+  auto dp = floor<days>(time_point);
+  return weekday(dp);
+}
+
+bool isWeekday(year_month_day date) {
+  weekday date_weekday = weekday(local_days(date));
+  return date_weekday != Saturday && date_weekday != Sunday;
+}
+
+template <typename FieldType>
+class SingleValueField : public CronField {
+ public:
+  explicit SingleValueField(FieldType value) : value_(value) {}
+
+  [[nodiscard]] bool matches(local_seconds time_point) const override {
+    return value_ == getFieldType<FieldType>(time_point);
+  }
+
+ private:
+  FieldType value_;
+};
+
+class NotCheckedField : public CronField {
+ public:
+  NotCheckedField() = default;
+
+  [[nodiscard]] bool matches(local_seconds) const override { return true; }
+};
+
+class AllValuesField : public CronField {
+ public:
+  AllValuesField() = default;
+
+  [[nodiscard]] bool matches(local_seconds) const override { return true; }
+};
+
+template <typename FieldType>
+class RangeField : public CronField {
+ public:
+  explicit RangeField(FieldType lower_bound, FieldType upper_bound)
+      : lower_bound_(std::move(lower_bound)),
+        upper_bound_(std::move(upper_bound)) {
+  }
```

Comment:
thats a good idea (especially since quartz numbers days differently)  I've added it in https://github.com/apache/nifi-minifi-cpp/commit/474cb1571b81b27a5e9871a63fea53cf30428e1c#diff-977a7d8d2ef0da59036a2e53f02f087ea0b5c962d8d3bad8a5c566d3842c322dR254-R255